### PR TITLE
Add spellcheck to test suite

### PR DESCRIPTION
**Pull request by leebyron**
Adds `cspell` and an extra word config, adding it to the test suite

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,11 +13,14 @@
     "url": "http://github.com/graphql/graphql-spec.git"
   },
   "scripts": {
-    "test": "spec-md spec/GraphQL.md > /dev/null",
+    "test": "npm run test:build && npm run test:spellcheck",
+    "test:build": "spec-md spec/GraphQL.md > /dev/null",
+    "test:spellcheck": "cspell 'spec/**/*.md'",
```
Comment (benjie) [inline package.json:18]: Might as well spellcheck the README and the RFCs too?

**Review by benjie — APPROVED** [review]

**Comment by benjie** [comment]
(My review assumes that you also did a search for the replaced text to ensure nothing escaped; I can't easily do that through the GitHub interface.)

File: package.json (hunk 1 above)
Comment (leebyron) [reply package.json:18]: README yes. RFCs have a zillion spurious errors (like flagging github usernames) so I'll exclude those

**Comment by leebyron** [comment]
> (My review assumes that you also did a search for the replaced text to ensure nothing escaped; I can't easily do that through the GitHub interface.)

Yep, I applied the changes manually rather than using any automated replacements to ensure nothing broke

**Pull request merged**